Sources/VLCSettingsSpecifierManager.swift

```diff
@@ -0,0 +1,107 @@
+    }
+    
+    var selectedIndex: IndexPath {
+        var index: Int
+        if let selectedItem = settingsStore.object(forKey: specifier.key()) {
+            index = items.index(of: selectedItem)
+        } else {
+            index = items.index(of: specifier.defaultValue())
+        }
+        return IndexPath(row: index, section: 0)
+    }
+    
+    @objc init(specifier: IASKSpecifier, settingsReader: IASKSettingsReader, settingsStore: IASKSettingsStore) {
+        self.specifier = specifier
+        self.settingsReader = settingsReader
+        self.settingsStore = settingsStore
+    }
+    
+    fileprivate func updateCollectionViewCellApparence(cell: VLCSettingsSheetCell, highlighted: Bool) {
+        var textColor = PresentationTheme.current.colors.cellTextColor
+        var tintColor = PresentationTheme.current.colors.cellDetailTextColor
+        
+        if highlighted {
+            textColor = PresentationTheme.current.colors.orangeUI
+            tintColor = PresentationTheme.current.colors.orangeUI
+        }
+        
+        cell.tintColor = tintColor
+        cell.name.textColor = textColor
+        cell.checkmark.isHidden = !highlighted
+    }
+}
+
+// MARK: VLCActionSheetDelegate
+
+extension VLCSettingsSpecifierManager: VLCActionSheetDelegate {
+    func headerViewTitle() -> String? {
+        return specifier.title()
+    }
+    
+    func itemAtIndexPath(_ indexPath: IndexPath) -> Any? {
+        return items[indexPath.row]
+    }
+    
+    func actionSheet(collectionView: UICollectionView, didSelectItem item: Any, At indexPath: IndexPath) {
+        if indexPath == selectedIndex {
+            return
+        }
+
+        guard let oldCell = collectionView.cellForItem(at: selectedIndex) as? VLCSettingsSheetCell,
+            let newCell = collectionView.cellForItem(at: indexPath) as? VLCSettingsSheetCell else {
+                assertionFailure("VLCSettingsSpecifierManager: VLCActionSheetDelegate: Cell is not a VLCSettingsSheetCell")
+                return
+        }
+
+        updateCollectionViewCellApparence(cell: oldCell, highlighted: false)
+        updateCollectionViewCellApparence(cell: newCell, highlighted: true)
+    }
+}
+
+// MARK: VLCActionSheetDataSource
+
+extension VLCSettingsSpecifierManager: VLCActionSheetDataSource {
+    func numberOfRows() -> Int {
+        return items.count
+    }
+    
+    func actionSheet(collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: VLCSettingsSheetCell.identifier, for: indexPath) as? VLCSettingsSheetCell else {
+            assertionFailure("VLCSettingsSpecifierManager: VLCActionSheetDataSource: Unable to dequeue reusable cell")
+            return UICollectionViewCell()
+        }
+        
+        let titles = specifier.multipleTitles()
+        cell.name.text = settingsReader.title(forStringId: titles?[indexPath.row] as? String)
```

Comment:
can we add a check for indexPath.row < titles.count? Otherwise we crash here and maybe as well that the object is of type String otherwise we fail here silently